builder/publish.go

```diff
@@ -4,95 +4,168 @@
-		}
-
 		branch, version, err := GetImageTagValues(tagMode)
 		if err != nil {
 			return err
 		}
 
 		imageName := schema.BuildImageName(tagMode, image, version, branch)
 
-		if err := ensureHandlerPath(handler); err != nil {
-			return fmt.Errorf("building %s, %s is an invalid path", imageName, handler)
-		}
+		if remoteBuilder != "" {
+			tempDir, err := os.MkdirTemp(os.TempDir(), "builder-*")
+			if err != nil {
+				log.Fatal(err)
+			}
+			defer os.RemoveAll(tempDir)
 
-		tempPath, buildErr := createBuildContext(functionName, handler, language, isLanguageTemplate(language), langTemplate.HandlerFolder, copyExtraPaths)
-		fmt.Printf("Building: %s with %s template. Please wait..\n", imageName, language)
-		if buildErr != nil {
-			return buildErr
-		}
+			tarPath := path.Join(tempDir, "req.tar")
+			fmt.Println(tarPath)
 
-		if shrinkwrap {
-			fmt.Printf("%s shrink-wrapped to %s\n", functionName, tempPath)
-			return nil
-		}
+			if err := shrinkWrap(imageName, handler, language); err != nil {
+				log.Fatal(err)
+			}
 
-		buildOptPackages, buildPackageErr := getBuildOptionPackages(buildOptions, language, langTemplate.BuildOptions)
+			if err := makeTar(buildConfig{Image: imageName}, path.Join("build", "context"), tarPath); err != nil {
+				log.Fatalf("Failed to create tar file: %s", err)
+			}
 
-		if buildPackageErr != nil {
-			return buildPackageErr
+			res, err := callBuilder(tarPath, remoteBuilder)
+			if err != nil {
+				log.Fatalf("Failed to call builder API: %s", err)
+			}
+			defer res.Body.Close()
 
-		}
+			data, _ := io.ReadAll(res.Body)
 
-		dockerBuildVal := dockerBuild{
-			Image:            imageName,
-			NoCache:          nocache,
-			Squash:           squash,
-			HTTPProxy:        os.Getenv("http_proxy"),
-			HTTPSProxy:       os.Getenv("https_proxy"),
-			BuildArgMap:      buildArgMap,
-			BuildOptPackages: buildOptPackages,
-			BuildLabelMap:    buildLabelMap,
-			Platforms:        platforms,
-			ExtraTags:        extraTags,
-		}
+			result := buildResult{}
+			if err := json.Unmarshal(data, &result); err != nil {
+				log.Fatalf("Failed to unmarshal build result: %s", err)
+			}
 
-		command, args := getDockerBuildxCommand(dockerBuildVal)
-		fmt.Printf("Publishing with command: %v %v\n", command, args)
+			if res.StatusCode != http.StatusOK && res.StatusCode != http.StatusAccepted {
+				log.Fatalf("Unable to build image %s: %s", imageName, result.Status)
+			}
 
-		task := v1execute.ExecTask{
-			Cwd:         tempPath,
-			Command:     command,
-			Args:        args,
-			StreamStdio: !quietBuild,
-		}
+			log.Printf("Success building image: %s", result.Image)
 
-		res, err := task.Execute()
-
-		if err != nil {
-			return err
-		}
-
-		if res.ExitCode != 0 {
-			return fmt.Errorf("[%s] received non-zero exit code from build, error: %s", functionName, res.Stderr)
+		} else {
+			pathToTemplateYAML := fmt.Sprintf("./template/%s/template.yml", language)
```

Comment:
I'm not sure why these code paths were duplicated, the shinkwrap code already exists - including the code to generate a build context.

Really, the change should just conditionally either:

1 - exec docker build / buildx build (as it does presently)
2 - make a HTTP call and return the logs into the console